### PR TITLE
ipn/localapi: send Tailscale version in ACME User-Agent

### PR DESCRIPTION
**Pull request by andrew-d**
Requested by a friend at Let's Encrypt.

cc @mcpherrinm

Signed-off-by: Andrew Dunham <andrew@tailscale.com>

### CODE REVIEW

File: ipn/localapi/cert.go [hunk 1]
```diff
@@ -185,7 +186,10 @@ func (h *Handler) getCertPEM(ctx context.Context, logf logger.Logf, traceACME fu
 	if err != nil {
 		return nil, fmt.Errorf("acmeKey: %w", err)
 	}
-	ac := &acme.Client{Key: key}
+	ac := &acme.Client{
+		Key:       key,
+		UserAgent: "tailscaled " + version.String(),
```
Comment (bradfitz) [inline ipn/localapi/cert.go:191]: This String isn't what you think it is

File: version/version.go [hunk 2]
```diff
@@ -23,7 +23,8 @@ var Long = ""
 
 // Short is a short version number for this build, of the form
 // "x.y.z" for builds stamped in the usual way (see
-// build_dist.sh in the root) or, for binaries built by hand with the                                                                            // go tool, it's like Long's dev form, but ending at the date part,
+// build_dist.sh in the root) or, for binaries built by hand with the
```
Comment (danderson) [inline version/version.go:26]: Spurious change? I can't tell what's different.

**Review by danderson — APPROVED** [review]

**Review by DentonGentry — APPROVED** [review]
Fine to submit in time for 1.30.

File: version/version.go (hunk 2 above)
Comment (mcpherrinm) [reply version/version.go:26]: The "split" github view hides it with line wrapping, but there's a missing newline here being inserted.  More obvious if you look at the current file in main: https://github.com/tailscale/tailscale/blob/27f36f77c364562bc1ca660c241291e23606e54d/version/version.go#L26

File: ipn/localapi/cert.go [hunk 3]
```diff
@@ -185,7 +186,10 @@ func (h *Handler) getCertPEM(ctx context.Context, logf logger.Logf, traceACME fu
 	if err != nil {
 		return nil, fmt.Errorf("acmeKey: %w", err)
 	}
-	ac := &acme.Client{Key: key}
+	ac := &acme.Client{
+		Key:       key,
+		UserAgent: "tailscaled " + version.Long,
```
Comment (bradfitz) [inline ipn/localapi/cert.go:191]: don't you want a slash instead of a space? https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent

**Review by bradfitz — APPROVED** [review]

File: ipn/localapi/cert.go (hunk 3 above)
Comment (mcpherrinm) [reply ipn/localapi/cert.go:191]: A slash would be consistent with the standard use of the User Agent header and other ACME clients, though from Let's Encrypt's perspective we don't care too much about the separator.

Comment (andrew-d) [reply ipn/localapi/cert.go]: Good call; switched to a slash.

**Review by bradfitz — APPROVED** [review]

**Pull request merged**